src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerStepExecution.java

```diff
@@ -43,7 +49,14 @@ public boolean start() throws Exception {
         client = nodeContext.connectToCloud();
 
         EnvironmentExpander env = getContext().get(EnvironmentExpander.class);
-        decorator = new ContainerExecDecorator(client, nodeContext.getPodName(), containerName, nodeContext.getNamespace(), env);
+        EnvVars globalVars = null;
+        Jenkins instance = Jenkins.getInstance();
+        DescribableList<NodeProperty<?>, NodePropertyDescriptor>globalNodeProperties = instance.getGlobalNodeProperties();
+        List<EnvironmentVariablesNodeProperty> envVarsNodePropertyList = globalNodeProperties.getAll(EnvironmentVariablesNodeProperty.class);
+        if ( envVarsNodePropertyList != null && envVarsNodePropertyList.size() != 0 ) {
+            globalVars = envVarsNodePropertyList.get(0).getEnvVars();
```

Comment:
I assume you need to iterate through `envVarsNodePropertyList` not just take the first item

Reply:
You shouldn't unless there are multiple GlobalNodeProperties in a config.